--- lib/jcsda-emc/spack-stack/stack/meta_modules.py	2023-12-12 02:44:14.068945+00:00
+++ lib/jcsda-emc/spack-stack/stack/meta_modules.py	2023-12-12 02:44:27.720853+00:00
@@ -715,11 +715,13 @@
         for compiler_version in compiler_dict[compiler_name]:
             if (
                 not compiler_name in python_dict.keys()
                 or not compiler_version in python_dict[compiler_name].keys()
             ):
-                logging.warn("No Python version found for compiler {compiler_name}@{compiler_version}, skipping Python modules")
+                logging.warn(
+                    "No Python version found for compiler {compiler_name}@{compiler_version}, skipping Python modules"
+                )
                 continue
             spec = python_dict[compiler_name][compiler_version]
             python_version = str(spec.version)
             logging.info(
                 "  ... configuring stack python interpreter {}@{} for compiler {}@{}".format(
