--- a/shapely/geos.py	2023-11-29 20:34:39.000000000 -0700
+++ b/shapely/geos.py	2023-11-29 20:35:39.000000000 -0700
@@ -88,14 +88,26 @@
         if len(geos_pyinstaller_so) >= 1:
             _lgeos = CDLL(geos_pyinstaller_so[0])
             LOG.debug("Found GEOS DLL: %r, using it.", _lgeos)
-    elif exists_conda_env():
-        # conda package.
-        _lgeos = CDLL(os.path.join(sys.prefix, 'lib', 'libgeos_c.so'))
+    # Spack might be using a barebone miniconda installation, in which case
+    # this elif block makes py-shapely look for libgeos_c.so in the miniconda
+    # install tree ... need to comment out so that spack's libgeos_c.so is found
+    #elif exists_conda_env():
+    #    # conda package.
+    #    _lgeos = CDLL(os.path.join(sys.prefix, 'lib', 'libgeos_c.so'))
     else:
-        alt_paths = [
-            'libgeos_c.so.1',
-            'libgeos_c.so',
-        ]
+        # Use geos installation if spack geos module is loaded
+        if 'GEOS_ROOT' in os.environ:
+            alt_paths = [
+                os.path.join(os.environ['GEOS_ROOT'], 'lib', 'libgeos_c.so'),
+                os.path.join(os.environ['GEOS_ROOT'], 'lib64', 'libgeos_c.so'),
+                os.path.join(os.environ['GEOS_ROOT'], 'lib', 'libgeos_c.so.1'),
+                os.path.join(os.environ['GEOS_ROOT'], 'lib64', 'libgeos_c.so.1'),
+            ]
+        else:
+            alt_paths = [
+                'libgeos_c.so.1',
+                'libgeos_c.so',
+            ]
         _lgeos = load_dll('geos_c', fallbacks=alt_paths)
 
     # ctypes.CDLL(None) internally calls dlopen(NULL), and as the dlopen
@@ -119,10 +131,12 @@
         else:
             _lgeos = CDLL(geos_whl_dylib)
             LOG.debug("Found GEOS DLL: %r, using it.", _lgeos)
-
-    elif exists_conda_env():
-        # conda package.
-        _lgeos = CDLL(os.path.join(sys.prefix, 'lib', 'libgeos_c.dylib'))
+    # Spack might be using a barebone miniconda installation, in which case
+    # this elif block makes py-shapely look for libgeos_c.so in the miniconda
+    # install tree ... need to comment out so that spack's libgeos_c.so is found
+    #elif exists_conda_env():
+    #    # conda package.
+    #    _lgeos = CDLL(os.path.join(sys.prefix, 'lib', 'libgeos_c.dylib'))
     else:
         if hasattr(sys, 'frozen'):
             try:
@@ -139,6 +153,9 @@
                 if hasattr(sys, '_MEIPASS'):
                     alt_paths.append(
                         os.path.join(sys._MEIPASS, 'libgeos_c.1.dylib'))
+        # Use geos installation if spack geos module is loaded
+        elif 'GEOS_ROOT' in os.environ:
+            alt_paths = [os.path.join(os.environ['GEOS_ROOT'], 'lib', 'libgeos_c.dylib')]
         else:
             alt_paths = [
                 # The Framework build from Kyng Chaos
